communicationmanager.go

```diff
@@ -180,9 +180,20 @@ func newCommunicationManager(cfg *config.Config) (cm *communicationManager, err
 
 	if cfg.Monitoring.MonitorConfig != nil {
 		if cm.resourcemonitor, err = resourcemonitor.New(*cfg.Monitoring.MonitorConfig, cm.iam, cm.unitConfig,
-			nil, cm.alerts, cm.monitorcontroller); err != nil {
+			nil, cm.alerts); err != nil {
 			return cm, aoserrors.Wrap(err)
 		}
+
+		go func() {
```

Comment:
I think it would be better to handle inside monitorController

Reply:
It doesn't fit into monitorcontroller interface. Currently, there is no better place for it.